extras/rya.streams/kafka/src/main/java/org/apache/rya/streams/kafka/processors/join/KeyValueJoinStateStore.java

```diff
@@ -77,30 +77,39 @@
     /**
      * This is the maximum value of a UTF-8 character.
      */
-    private static final String END_RANGE_SUFFIX = new String(new byte[] { (byte) 0XFF }, Charsets.UTF_8);
+    private static final String END_RANGE_SUFFIX = new String(new byte[] { (byte) 0xFF }, Charsets.UTF_8);
+
+    /**
+     * Indicates where the end of the join variables occurs.
+     */
+    private static final String JOIN_VAR_END_MARKER = new String("~!^~".getBytes(Charsets.UTF_8), Charsets.UTF_8);
```

Comment:
I'm not sure that the marker should be in the visually renderable byte ranges since, while very unlikely, they could legitimately appear in the data that is being indexed.

Reply:
Done. Changed to 0x06,0x10,0x03,0x04.